### PR TITLE
add CorrectionManager

### PR DESCRIPTION
**Pull request by harrypuuter**

### CODE REVIEW

File: code_generation/analysis_template.cxx [hunk 1]
```diff
@@ -108,6 +110,8 @@ int main(int argc, char *argv[]) {
 
     // start an onnx session manager
     OnnxSessionManager onnxSessionManager;
+    // start a correction manager
+    CorrectionManager correctionManager;
```
Comment (nshadskiy) [inline code_generation/analysis_template.cxx:114]: I think we should also add this in `analysis_template_friends.cxx`. For example for fake factors also a correctionlib file is used.

File: src/utility/CorrectionMananger.cxx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#include <string>
+#include <unordered_map>
+#include <nlohmann/json.hpp>
+#include <fstream>
+/// namespace used for the CorrectionManager
+namespace correctionManager {
+/**
+ * @brief Load a correctionlib correction from a file and return it. This
+ * function works for most corrections found in correctionlib files (the default
+ * type for corrections is correction::Correction) . If the requested Correction
```
Comment (ralfschmieder) [inline src/utility/CorrectionMananger.cxx:13]: ).
sometimes Correction sometimes correction

File: src/utility/CorrectionMananger.cxx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ * @param corrName The name of the correction to load
+ * @return const correction::Correction*
+ */
+const correction::Correction *
+CorrectionManager::loadCorrection(const std::string &filePath,
+                                  const std::string &corrName) {
+    auto filePath_it = correction_map.find(filePath);
+    if (filePath_it == correction_map.end()) {
+        Logger::get("CorrectionManager")
+            ->debug("CorrectionFile {} not loaded yet, adding it...", filePath);
```
Comment (ralfschmieder) [inline src/utility/CorrectionMananger.cxx:27]: mabye:
adding it to CorrectionManager

**Review by ralfschmieder — APPROVED** [review]
minor suggestions.
Otherwise looks good and was also tested for Wh_htt

**Pull request merged**